### PR TITLE
[MM-30932] Adding support for Rotator tool in cluster upgrades.

### PR DESCRIPTION
**Pull request by stylianosrigas**
#### Summary
- Adding support for Rotator tool in cluster upgrades
- This is pending [PR](https://github.com/mattermost/rotator/pull/1)

Once rotator gets merged, the go.mod will be updated here.

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-30931
- https://mattermost.atlassian.net/browse/MM-30932
- 
#### Release Note
```release-note
- Adding support for Rotator tool in cluster upgrades
```

### CODE REVIEW

File: go.mod
Comment (gabrieljackson) [inline go.mod:18]: Can't wait to get this in! Let's hold off a bit and get the rotator PR merged first so we can point our module at that commit (or tag if we decide that is worth it).

Comment (stylianosrigas) [reply go.mod:18]: Yeap that will be blocked until the rotator gets merged and we release v0.1.0. I just created the PR to track all other changes ;)

File: cmd/cloud/cluster.go
Comment (angeloskyratzakos) [inline cmd/cloud/cluster.go:75]:
```suggestion
	clusterUpgradeCmd.Flags().Int("max-scaling", 5, "The maximum number of nodes to rotate every time. If the number is bigger than the number of nodes, then the number of nodes will be the maximum number.")
```

File: go.mod
Comment (stylianosrigas) [reply go.mod]: This should be ready to review now :)

File: go.mod
Comment (gabrieljackson) [inline go.mod:31]: Hmm, this is getting ahead of the operator imports. Do we know if we want to include these changes?

Comment (stylianosrigas) [reply go.mod]: You are right. These changes were brought by the rotator package that was using newer versions. A new rotator release was cut to use the same versions as the provisioner and it is good now.

**Review by gabrieljackson — APPROVED** [review]
Awesome work!

**Review by angeloskyratzakos — APPROVED** [review]
LGTM

**Pull request merged**
